### PR TITLE
debug-guide.md added memory statistics command

### PR DESCRIPTION
**Pull request by rob-deutsch**
This would've been valuable to me while investigating #5530.

I discovered that more debug commands are also available, but I don't know enough about them to add them myself (e.g. `debug/pprof-mutex`, `debug/metrics/prometheus`).

### CODE REVIEW

**Comment by rob-deutsch** [comment]
Thanks @schomatis, resubmitted with suggested additions.

File: docs/debug-guide.md [hunk 1]
```diff
@@ -84,6 +86,10 @@ about `go tool pprof`. My go-to method of analyzing these is to run the `web`
 command, which generates an SVG dotgraph and opens it in your browser. This is
 the quickest way to easily point out where the hot spots in the code are.
 
+### Analysing vars and memory statistics
```
Comment (schomatis) [inline docs/debug-guide.md]: nit: could you also add an entry in the "Table of Contents" above pointing here?

Comment (rob-deutsch) [reply docs/debug-guide.md]: Drat, my bad, will do

**Comment by rob-deutsch** [comment]
... I also noticed (and fixed) the inconsistent spelling of analyse/analyze, just to make it a tad neater.

**Comment by schomatis** [comment]
> ... I also noticed (and fixed) the inconsistent spelling of analyse/analyze, just to make it a tad neater.

I think they are both right but "analyze" is the accepted American spelling.

**Comment by rob-deutsch** [comment]
In the case of ambiguity I tend to prefer British English, but some after some Googling I learned that US English is generally accepted for this kind of purpose.

Resubmitted with US English and a small frown of disapproval 😉

**Comment by Kubuxu** [comment]
There is also a much simpler option for non-linux folks to download most of the info: going to `localhost:5001/debug/` and doing `Save as` on the links.

Not blocking this PR in any regard, just mentioning it as I discovered it recently.

**Comment by rob-deutsch** [comment]
I'm getting a 404 error on `localhost:5001/debug/`. That sounds like a handy feature though, can we integrate that to master?

**Comment by Kubuxu** [comment]
Sorry: `localhost:5001/debug/pprof/`.

**Pull request merged**